docs/docs/tutorial/part-7/index.mdx

```diff
@@ -0,0 +1,492 @@
+---
+
+import { Announcement, LinkButton } from 'gatsby-interface'
+import Collapsible from '@components/collapsible'
+import { MdArrowForward } from 'react-icons/md'
+
+## Introduction
+
+In Part 3, you used `gatsby-plugin-image` to add static images to your home page. Now that you've worked a bit more with Gatsby's data layer, it's time to revisit `gatsby-plugin-image`. This time, you'll learn how to add dynamic images to your site.
+
+In this part of the Tutorial, you'll use the dynamic `GatsbyImage` component to add hero images to each of your blog posts.
+
+By the end of this part of the Tutorial, you will be able to:
+
+- Use the `GatsbyImage` component to create images dynamically from data.
+
+## Add hero images to blog post frontmatter
+
+Many blog sites include a hero image at the top of each post. These images are usually large, high-quality photos that are meant to grab the reader's attention and make them want to stay on the page longer.
+
+The steps below will help you find and download some photos for your hero images and add them to the frontmatter for each of your blog posts.
+
+1. Start by organizing the `blog` directory with all your MDX posts. First, create a new subdirectory in your `blog` folder for each post. Then, rename each of your `.mdx` files to `index.mdx` (to prevent the routes from ending up with a duplicated path parameter, like `blog/my-post/my-post/`).
+    * For example, a post at `blog/my-first-post.mdx` would move to `blog/my-first-post/index.mdx`. Similarly, a post at `blog/another-post.mdx` would move to `blog/another-post/index.mdx`.
+
+<Announcement style={{marginBottom: "1.5rem"}}>
+
+**Note:** After you move or rename your `.mdx` files, you'll need to stop and restart your local development server for your changes to be picked up.
+
+</Announcement>
+
+![Diagram of moving each MDX file into its own subdirectory in the blog folder](./reorganize-blog-directory.png)
+
+2. Use a website like [Unsplash](https://unsplash.com) to find some pretty, freely usable images. For best results, choose photos with a landscape (horizontal) orientation, since those will fit on your screen more easily.
+    ![A screenshot of the Unsplash home page](./unsplash.png)
+
+2. When you've found a photo that you like, download it and add it to subdirectory for one of your blog posts. Continue downloading photos until you have a different hero image for each post.
+
+3. Next, add some additional frontmatter fields to each of your blog posts:
+    * `hero_image`: the relative path to the hero image file for that post
+    * `hero_image_alt`: a short description of the image, to be used as alternative text for screen readers or in case the image doesn't load correctly
+    * `hero_image_credit_text`: the text to display to give the photographer credit for the hero image
+    * `hero_image_credit_link`: a link to the page where your hero image was downloaded from
+
+```mdx:title=blog/my-first-post/index.js
```

Comment:
File endings for the examples here are `.js` while should be `.mdx`

Reply:
Good catch, thanks! Fixed :)